src/backends/llvm.ts

```diff
@@ -83,42 +133,75 @@ function emit_lookup(emitter: LLVMEmitter, emit_extern: (name: string, type: Typ
   }
 }
 
+function emit_extern(name: string, type: Type): llvm.Value {
+  if (_is_fun_type(type)) {
+    // The extern is a function. Wrap it in the clothing of our closure
+    // format (with no environment).
+    // TODO
+    throw "not implemented"
+  } else {
+    // An ordinary value. Just look it up by name.
+    // TODO
+    throw "not implemented"
+  }
+}
+
+// mostly a copy of emitter function
+function emit(emitter: LLVMEmitter, tree: ast.SyntaxNode): llvm.Value {
+  return emitter.emit_expr(tree, emitter);
+}
+
 ///////////////////////////////////////////////////////////////////
 // End Emit Functions & Redundant Funcs
 ///////////////////////////////////////////////////////////////////
 
+/**
+ * Create an alloca with the provided name in the entry block of the provided function
+ */
+function create_entry_block_alloca(func: llvm.Function, type: llvm.Type, name: string): llvm.Value {
+  // create builder and position after func's first instruction
+  let builder: llvm.Builder = llvm.Builder.create();
```

Comment:
That's a good point—I had totally forgotten that I did that for the other backends, using C89-style declarations at the top of the function. Fortunately, it's not *too* hard to do since the compiler analysis has already identified all the bound (local) variables for each function.